src/Okta.Sdk/CollectionClient{T}.cs

```diff
@@ -34,8 +35,8 @@ public sealed class CollectionClient<T> : ICollectionClient<T>
         }
 
         /// <inheritdoc/>
-        public IAsyncEnumerator<T> GetEnumerator()
-            => new CollectionAsyncEnumerator<T>(_dataStore, _initialRequest, _requestContext);
+        public IAsyncEnumerator<T> GetAsyncEnumerator(CancellationToken cancellationToken = default)
+        => new CollectionAsyncEnumerator<T>(_dataStore, _initialRequest, _requestContext);
```

Comment:
Here the `cancellationToken` is being thrown away. But it seems like the right fix is to update `CollectionAsyncEnumerator` (and the `PagedCollectionEnumerator` it contains) to accept the cancellation token in their constructors instead of through `MoveNext()`. Is there a bug tracking this work?

It could be that I'm missing something but to me it doesn't seem like that change is a ton of work. Maybe it is worth doing this now?

Reply:
How to use this token is still unclear to me. The `IAsyncEnumerable` interface forces you to accept a cancellation token in the `GetAsyncEnumerator()` method which is not clear yet how to use it inside of the enumerator:

```
    public interface IAsyncEnumerable<out T>
    {
        IAsyncEnumerator<T> GetAsyncEnumerator(CancellationToken cancellationToken = default);
    }
```
[I created an issue on MS docs repo ](https://github.com/dotnet/docs/issues/17036)to add more details about what the recommended practice is here, since I couldn't find any good content that explains this clearly.

Reply:
I was thinking that we would use it here, in the call to `IDataStore.GetArrayAsync<T>(...)` that `PagedCollectionEnumerator<T>.MoveNextAsync(...)` makes. My interpretation of the interface is that passing the `CancellationToken` to the enumerator's constructor should be equivalent to passing it to any of the enumerator's methods when they are called.

https://github.com/okta/okta-sdk-dotnet/blob/876fb997886a23785d8827d4f637ea6bbde02a14/src/Okta.Sdk/PagedCollectionEnumerator%7BT%7D.cs#L50-L53

Reply:
Oh I see what you mean, and yes, you're right. This needs to be fixed. Thanks for the heads up.